File: Python/Kyu 5/sum of pairs.py
# https://www.codewars.com/kata/sum-of-pairs


def sum_pairs(ints, s=0):
    intsSet = set(ints)
    indexesa = []
    for i in intsSet:
        result = s - i
        if result in intsSet:
            indexesa += [x for x, v in enumerate(ints) if v == i]
    indexesa = list(sorted(indexesa))
    l2 = []
    for i in indexesa:
        for x in range(len(indexesa)):
            if ints[i] + ints[indexesa[x]] == s and i != indexesa[x] and i < indexesa[x]:
                if len(l2) == 0:
                    l2 = [i, indexesa[x]]
                elif indexesa[x] < l2[1]:
                    l2 = [i, indexesa[x]]
    if len(l2) == 0:
        return None
    else:
        l2 = list(sorted(l2))
        return [ints[l2[0]], ints[l2[1]]]

File: JavaScript/kyu_5/sum_of_pairs.js
// https://www.codewars.com/kata/54d81488b981293527000c8f/javascript

const sumPairs = function (ints, s) {
    if (ints.length < 2) return undefined;
    let intSet = new Set();
    for (let i = 0; i < ints.length; i++) {
        let needed = s - ints[i];

        if (intSet.has(needed)) {
            return [needed, ints[i]];
        }
        intSet.add(ints[i]);
    }
    return undefined;
};
